unittests/tools/test_qualys_parser.py

```diff
@@ -70,7 +70,7 @@ def parse_file_with_multiple_vuln_has_multiple_findings(self):
             self.assertEqual(finding_cvssv3_vector.cvssv3,
                             "CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:C/C:H/I:H/A:H")
             self.assertEqual(
-                finding_cvssv3_vector.severity, "Critical"
+                finding_cvssv3_vector.severity, "High"
```

Comment:
The need for this line to change tells me that this will be unexpected behavior for qualys users. What used to be `Critical` will now to be demoted to a `High`...

I looked at your first commit, and see that it is fixing the issue described in the pull request. The move to using CVSS to attempt to be "smarter than the vendor" is a slippery path that has bit us before as @mtesauro has mentioned. I think reverting to your first commit, and then adding a unit test for that change would be the best path to success with this PR

Reply:
@Maffooch

This line had to change because for that vulnerability the Finding severity didn't match the CVSS severity. So again, the question is, what should DD use first?
- The severity explicitly given by the vendor, or
- The mapped value from the CVSS score to severity.

In other words, that line changed because this PR changes the priority in how to populate the finding severity: before it was "CVSS Severity" then "real severity", this PR uses: "real severity" and if not available fallback to "CVSS Severity".

In the past, DD went for this PR's approach: https://github.com/DefectDojo/django-DefectDojo/issues/6248#issuecomment-1164293652. If you think this is a mistake, maybe the Veracode SCA parser also needs to be changed?

For context:
Qualys calls this the "CVSS Severity" (https://blog.qualys.com/qualys-insights/2022/10/10/in-depth-look-into-data-driven-science-behind-qualys-trurisk) and it might not always match the finding severity (https://success.qualys.com/support/s/article/000002759). Extracts:
<img width="1233" alt="image" src="https://github.com/DefectDojo/django-DefectDojo/assets/7769945/e19b7cbe-d09d-4004-9bf0-27a1a3bc5e5b">
<img width="856" alt="image" src="https://github.com/DefectDojo/django-DefectDojo/assets/7769945/35e7b2c9-aa28-409d-81a0-c939a5dccb93">

Reply:
@Maffooch 
I think there is a misunderstanding: the current version of the Qualys parser is the one that tries to be "smarter than the vendor", as it uses the CVSS, whereas this PR is simply attempts to use qualys' severity. In other words, we all agree with @mtesauro: DD should use the already available 'Severity' field by the vendor, before trying CVSS to fill DD's 'severity' field. Similar to other parsers in DD: https://github.com/DefectDojo/django-DefectDojo/issues/6248#issuecomment-1164293652....TBH, I've never seen a qualys vulnerability without 'Severity' field, so I'd be happy to remove the fallback to CVSS altogether, as it will probably never be used.

FYI:
The change of severity in the other unittest, is because of this mismatch between Severity and CVSS Severity:
<img width="863" alt="image" src="https://github.com/DefectDojo/django-DefectDojo/assets/7769945/46ae6351-5b7e-4b88-8b4b-bdb09d6f249b">

A 'CVSSv3 severity' translates to 'Critical' (old code), but the 'vendor provided severity' is 4 in a scale 1-5, so it translates to 'High' (this PR code).

Reply:
You're right. I was confusing the CSV parser and the XML one. I agree with removing the CVSS fallback altogether. That way it would match the CSV parser

@mtesauro what do you think about that?

Reply:
@Maffooch @nv-pipo 
I agree on removing the CVSS fallback completely on this PR given what's been said on this thread (including what Qualys says about how they handle severity / CVSS - it seems the most sensible way forward.